File: allure-testng/src/test/java/io/qameta/allure/examples/testng/AllureSimpleTest.java
package io.qameta.allure.examples.testng;

import io.qameta.allure.Step;
import io.qameta.allure.model.Status;
import org.testng.annotations.Test;

import static io.qameta.allure.Allure.step;

public class AllureSimpleTest {

    @Test(description = "allureSimpleTest displayName")
    public void allureSimpleTest() {
        step("Simple step");
        step("Simple step with status", Status.FAILED);
        step("Simple lambda step", () -> {
            step("Simple step inside lambda step");
        });
        simpleTestMethod("method parameter");
    }

    @Step("Simple test method with step annotation")
    public void simpleTestMethod(String param) {
        step("Method parameter: " + param);
        step("Simple step inside test method");
    }

}

File: kotlin-junit5-gradle-kts/src/test/kotlin/io/qameta/allure/examples/junit5/AllureSimpleTest.kt
package io.qameta.allure.examples.junit5

import io.qameta.allure.Allure.step
import io.qameta.allure.examples.kotlin.AllureHelpers.step
import io.qameta.allure.Step
import io.qameta.allure.model.Status
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class AllureSimpleTest {

    @Test
    @DisplayName("allureSimpleTest displayName")
    fun allureSimpleTest() {
        step("Simple step")
        step("Simple step with status", Status.FAILED)
        step("Simple lambda step") {
            step("Simple step inside lambda step")
        }
        simpleTestMethod("method parameter")
    }

    @Step("Simple test method with step annotation")
    fun simpleTestMethod(param: String) {
        step("Method parameter: $param")
        step("Simple step inside test method")
    }

}